packages/plugin-ext/src/plugin/uri-ext.ts

```diff
@@ -0,0 +1,65 @@
+// *****************************************************************************
+// Copyright (C) 2024 STMicroelectronics.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import * as theia from '@theia/plugin';
+import {
+    UriExt,
+    PLUGIN_RPC_CONTEXT, PluginInfo, UriMain
+} from '../common/plugin-api-rpc';
+import { RPCProtocol } from '../common/rpc-protocol';
+import { Disposable, URI } from './types-impl';
+import { UriComponents } from '../common/uri-components';
+
+export class UriExtImpl implements UriExt {
+
+    private static handle = 0;
+    private handles = new Set<string>();
```

Comment:
This set does not store handles, it stores the ids of the extensions that have already registered a handler. `registeredExtensions` maybe?